pageserver/src/tenant/timeline.rs

```diff
@@ -1860,6 +1867,31 @@ impl Timeline {
         Ok(lease)
     }
 
+    #[instrument(skip(_ctx), ret(Debug), err(Debug))]
+    pub(crate) fn lease_standby_horizon(
+        &self,
+        lease_id: String,
+        lsn: Lsn,
+        _ctx: &RequestContext,
+    ) -> anyhow::Result<SystemTime> {
+        if self
+            .feature_resolver
+            .evaluate_boolean("standby-horizon-leases-track-disable")
+            .is_ok()
+        {
+            // Use feature-flagging as a fail-safe in case something is wrong with the data structure (memory consumption, etc.)
+            return Ok(SystemTime::now()
+                .checked_add(Duration::from_secs(5 * 60))
+                .unwrap());
+        }
+        let applied_gc_cutoff_lsn =
+            todo!("think about boundary conditions? we didn't have any before though");
```

Comment:
the lsn leases do a bunch of checks around gc cutoffs

The existing standby_horizon mechanism doesn't perform any such checks